cowprotocol/services
# Support simulating using the pending timestamp with enso

# Description
When we simulate a transaction in the driver we only care whether it would work on the `pending` block and not on the `latest` block. Of course we can't magically predict the entire state of the blockchain on the next block but at least we could catch problems that arise of the next block number, timestamp and basefee.
This is pretty easy to do with `tenderly` and `web3` but unfortunately the `enso` simulator does not really support simulating on the `pending` block.

To get closer to the desired behavior I opened a [PR](https://github.com/EnsoFinance/temper/pull/27) (not yet merged) on the enso simulator which allows users to provide the timestamp they want to simulate on and changed our code to produce the needed timestamp in the `enso` simulator.

# Changes
- added `block_timestamp` to the `enso` dto
- added `current_block_stream` and `network_block_interval` to `enso` simulator to get the needed timestamp
- added plumbing to configure `network_block_interval` in a config file
- extended `FetchBlock.sol` to also return the blocks timestamp

## How to test
Did a manual test locally. Created a test order and checked that the enso simulator can handle the timestamp.
Unfortunately the `enso` team did not build a new docker image so I also tested that our code doesn't break with the old `temper` version so we should be able to merge this PR without any issues. Will contact the team to request a new build from them.

Fixes #2174
